slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java

```diff
@@ -3124,4 +3183,46 @@ private static <T> String getJsonWithGsonAnonymInnerClassHandling(List<T> tList)
         return GSON_ANONYM_INNER_CLASS_INIT_OUTPUT.equals(json) ? GSON.toJson(new ArrayList<>(tList)) : json;
     }
 
+    private static EntityMetadata[] prepareEntities(EntityMetadata[] entities) {
+        List<EntityMetadata> updatedEntities = new ArrayList<EntityMetadata>();
+        for (EntityMetadata entity : entities) {
+            entity = prepareEntityMetadata(entity);
+            updatedEntities.add(entity);
+        }
+        EntityMetadata[] entityArray = new EntityMetadata[updatedEntities.size()];
+        entityArray = updatedEntities.toArray(entityArray);
+
+        return entityArray;
+    }
+
+    private static EntityMetadata prepareEntityMetadata(EntityMetadata entity) {
+        if (entity.getEntityPayload().getFileFields() != null) {
+            String json = GSON.toJson(entity.getEntityPayload().getFileFields(),
+                    EntityPayload.FileFields.class);
+            JsonElement fields = GSON.fromJson(json, JsonElement.class);
+            entity.getEntityPayload().setFields(fields);
+            entity.getEntityPayload().setFileFields(null);
+        } else if (entity.getEntityPayload().getTaskFields() != null) {
+            String json = GSON.toJson(entity.getEntityPayload().getTaskFields(),
+                    EntityPayload.TaskFields.class);
+            JsonElement fields = GSON.fromJson(json, JsonElement.class);
+            entity.getEntityPayload().setFields(fields);
+            entity.getEntityPayload().setTaskFields(null);
+        } else if (entity.getEntityPayload().getIncidentFields() != null) {
+            String json = GSON.toJson(entity.getEntityPayload().getIncidentFields(),
+                    EntityPayload.IncidentFields.class);
+            JsonElement fields = GSON.fromJson(json, JsonElement.class);
+            entity.getEntityPayload().setFields(fields);
+            entity.getEntityPayload().setIncidentFields(null);
+        } else if (entity.getEntityPayload().getContentItemFields() != null) {
+            String json = GSON.toJson(entity.getEntityPayload().getContentItemFields(),
+                    EntityPayload.ContentItemFields.class);
+            JsonElement fields = GSON.fromJson(json, JsonElement.class);
+            entity.getEntityPayload().setFields(fields);
+            entity.getEntityPayload().setContentItemFields(null);
+        }
+
+        return entity;
+    }
```

Comment:
👾 suggestion(non-blocking): I'm not familiar enough in `java` to know if this is possible or a good idea, but could we export this from the models implementation file?

```
slack-api-model/src/main/java/com/slack/api/model/EntityMetadata.java
```